cmd/tikv-ctl/src/main.rs

```diff
@@ -868,18 +891,14 @@ fn flashback_whole_cluster(
                 Ok(res) => {
                     if let Err(key_range) = res {
                         // Retry specific key range to prepare flashback.
-                        let retry_stores_leader = load_leaders_to_each_store(
-                            &pd_client,
-                            key_range.get_start_key().to_vec(),
-                            key_range.get_end_key().to_vec(),
-                        );
-                        // Need to update `stores_leader` to replace stale key range.
-                        for (store_id, leaders) in retry_stores_leader {
-                            let regions = stores_leader
-                                .entry(store_id)
-                                .or_insert_with(HashMap::default);
-                            regions.extend(leaders);
-                        }
+                        let key_range = load_key_range(&pd_client, key_range.start_key.clone(), key_range.end_key.clone());
+                        key_range.into_iter().for_each(|((start_key, end_key), (region_id, store_id))| {
+                            // Need to update `prepare_key_range` to replace stale key range.
+                            let mut check = prepare_key_range.write().unwrap();
```

Comment:
Do we need to lock it inside the `for_each`? Could we just lock it once outside the loop?